playwright-e2e/dsm/component/filters/sections/customize-view.ts

```diff
@@ -39,28 +40,29 @@ export class CustomizeView {
   }
 
   private async selectOrDeselect(columnName: string, deselect = false): Promise<void> {
-    const columnXPath = this.columnsGroupXPath + this.columnPathXPath(columnName);
-    const column = this.page.locator(columnXPath);
-
-    if (!deselect) {
-      if (!(await this.isChecked(column))) {
-        await this.page.locator(columnXPath).click()
-      }
-    } else if (await this.isChecked(column)) {
-        await this.page.locator(columnXPath).click()
+    const checkbox = this.columnCheckbox(columnName);
+    const checked = await checkbox.isChecked();
+    if (deselect) {
+      if (checked) {
+        return checkbox.check();
       }
+      return;
+    }
+    if (!checked) {
+      return checkbox.check();
+    }
   }
 
   private async openColumnsGroup(opts: { nth?: number } = {}): Promise<void> {
     const { nth } = opts;
     const columnsGroupButton = this.columnsGroupButton({nth});
-    !(await this.isExpanded(columnsGroupButton)) && (await columnsGroupButton.click());
+    !(await this.isExpanded(columnsGroupButton)) && (await columnsGroupButton.locator('xpath=//*[@class="caret"]').click());
   }
 
   private async closeColumnsGroup(opts: { nth?: number } = {}): Promise<void> {
     const { nth } = opts;
     const columnsGroupButton = this.columnsGroupButton({nth});
-    (await this.isExpanded(columnsGroupButton)) && (await columnsGroupButton.click());
+    (await this.isExpanded(columnsGroupButton)) && (await columnsGroupButton.locator('xpath=//*[@class="caret"]').click());
```

Comment:
Looks for button's child span element to click in order to expand or close columns group dropdown.